src/main/java/io/confluent/connect/jdbc/dialect/OracleDatabaseDialect.java

```diff
@@ -72,6 +76,21 @@ protected String checkConnectionQuery() {
     return "SELECT 1 FROM DUAL";
   }
 
+  @Override
+  protected boolean maybeBindPrimitive(
+      PreparedStatement statement,
+      int index,
+      Schema schema,
+      Object value
+  ) throws SQLException {
+    if (schema.type() == Type.STRING) {
+      statement.setNString(index, (String) value);
```

Comment:
Is the setNString method backwards compatible with the setString method? Does it target the same types?

Reply:
> Is the setNString method backwards compatible with the setString method? Does it target the same types?

good question. I did test it against varchar, varchar2, nvarchar, nvarchar2 and clob columns, works fine. Didn't test long,  but long is not recommended by oracle.